### ISSUE TITLE
[TEALIUM SWIFT BUG]: Retain Cycle in TealiumConnectivity

### ISSUE THREAD

**Issue by tyler8691** [Tealium/tealium-swift#106]
**Your Environment**
*Fill out as appropriate.*
iOS Version(s) [e.g. 12.1]: 13
Xcode version [Xcode> About Xcode e.g. 10.1 (10B61)]: 11
Swift version [e.g. 4.2]: 5
Tealium Swift library version [e.g. 1.6.7]: 1.6.7
Dependency manager [None/Carthage/CocoaPods]: Carthage
Dependency manager version:  0.34.0

**Describe the bug**
When viewing the memory graph debugger, there is a retain cycle in `TealiumConnectivity`. `TealiumConnectivity` retains a `TealiumRepeatingTimer`, and the timer retains `TealiumConnectivity` through the `eventHandler.context`.

**Does a workaround exist?**
N/A

**To Reproduce**
Run an app with the Tealium SDK and capture the memory debugger output after firing some analytics events.

**Expected behavior**
No retain cycles should exist.

**Screenshots**
![Screen Shot 2019-10-31 at 1 30 38 PM](https://user-images.githubusercontent.com/3145469/67971504-46791880-fbe3-11e9-995d-fda5e1139b53.png)

**Additional context**
N/A

**Comment by christinasund** [comment]
@tyler8691 Thanks for bringing this to us. We just released a beta version where we did make some changes around the connectivity module and repeating timer. [1.8.0-beta1](https://github.com/Tealium/tealium-swift/releases/tag/1.8.0-beta1) This _should_ resolve the retain cycle you are seeing, but we haven't yet confirmed. We will look into this more and get back to you. In the meantime, if you would like to give the beta a try let us know what you think!
